src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -623,7 +623,8 @@ static TransactionProperties DecideTransactionPropertiesForTaskList(RowModifyLev
 																	List *taskList,
 																	bool
 																	exludeFromTransaction);
-static void StartDistributedExecution(DistributedExecution *execution);
+static void StartDistributedExecution(DistributedExecution *execution,
+									  bool localExecutionSupported);
```

Comment:
It is not the best solution to pass this as another flag to `StartDistributedExecution` but I thought that saving this in `DistributedExecution` could be confusing for readers, as we should always consider the global local execution state.

Reply:
For me, seeing this was more confusing. `localExecutionSupported` is different from the global local execution state. It seemed unusual to see a boolean parameter next to a struct which holds all the required information for the execution.

Reply:
opened an issue for further discussions https://github.com/citusdata/citus/issues/5637